app/__init__.py

```diff
@@ -228,6 +229,10 @@ def create_app(runtime_environment):
     flask_app.config["SQLALCHEMY_ENGINE_OPTIONS['pool_size']"] = app_config.db_pool_size
     flask_app.config["SQLALCHEMY_ENGINE_OPTIONS['pool_timeout']"] = app_config.db_pool_timeout
     flask_app.config["SQLALCHEMY_ENGINE_OPTIONS['pool_pre_ping']"] = True
+    if runtime_environment == RuntimeEnvironment.SERVER and app_config.db_statement_timeout > 0:
+        flask_app.config["SQLALCHEMY_ENGINE_OPTIONS['connect_args']"] = {
+            "options": f"-c statement_timeout={app_config.db_statement_timeout}"
+        }
```

Comment:
The runtime environment check should be moved into `config.py` (see my other comment) so that the RuntimeEnvironment-based logic is handled in one place. Then this can just be:
```suggestion
    flask_app.config["SQLALCHEMY_ENGINE_OPTIONS['connect_args']"] = {
        "options": f"-c statement_timeout={app_config.db_statement_timeout}"
    }
```